sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionProcessorImpl.java

```diff
@@ -248,7 +271,39 @@ public Mono<Void> run(CancellationToken cancellationToken) {
                                     }).flatMap(values -> Flux.empty());
                             }
                         }
-                        break;
+                        case PARSING_ERROR:
+                            if (this.unparseableDocumentRetries.incrementAndGet() == 1) {
```

Comment:
Though its atomic integer, but to avoid any concurrency issues, how about checking > 0?